iquip/apps/stage.py

```diff
@@ -0,0 +1,163 @@
+"""Module for stage controller."""
+
+import functools
+import logging
+from typing import Callable, Dict, Optional, Tuple
+
+from sipyco.pc_rpc import Client
+from PyQt5.QtCore import QObject, Qt, pyqtSignal, pyqtSlot
+
+logger = logging.getLogger(__name__)
+
+RPCTargetInfo = Tuple[str, int, str]  # ip, port, target_name
+
+def use_client(function: Callable[..., None]) -> Callable[..., None]:
+    """Decorator which substitutes a string key to a client object.
+
+    If the key does not exist, function is not called at all.
+    If an OSError occurs while running function, the RPC client is closed and
+        removed from the client dictionary. 
+    
+    Args:
+        function: Decorated function. It should take a Client object as the
+            first argument.
+    """
+    @functools.wraps(function)
+    def wrapped(self: "StageManager", key: str, *args, **kwargs):
+        """Translates a string key to a client.
+        
+        Args:
+            key: String key for identifying the client.
+        """
+        client = self._clients.get(key, None)  # pylint: disable=protected-access
+        if client is None:
+            logger.error("Failed to get client %s.", key)
+            return
+        try:
+            function(self, client, *args, **kwargs)
+        except OSError:
```

Comment:
Why did you check only `OSError`?
According to [Client](https://github.com/m-labs/sipyco/blob/master/sipyco/pc_rpc.py#L70), there seems to be possibilities that `ValueError` and `AttributeError` are raised.

Reply:
That's correct! I guess it would be good to catch them all. Even though `AttributeError` is the programmer's responsibility, catching it will helpful for stability.